### PR TITLE
feat: Create bookings (1/2)

### PR DESCRIPTION
**Pull request by atoff**
First half (pre-payment) functionality to create a booking

TODO:
- [x] #37 Implement ability to resume booking
- [x] #39 "Clicking book doesn't auto set production for booking"
- [x] Override global scroll-to behaviour inside Booking Stages router links (check out router/index.js scrollBehavior function)
- [x] Check for any more TODOS
- [x] Test book.vue and bookingStages.js
- [x] Fix ProductionCarousel test to use runApolloQuery
- [x] Need to do some more testing of landing on incorrect stages, etc
- [x] Implement auth middleware on booking pages

### CODE REVIEW

File: tests/unit/views/booking/Book.spec.js [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+  it('has booking navigation', () => {
+    let bookingNavigation = bookingComponent.findComponent(BookingNavigation);
+    expect(bookingNavigation.exists()).to.be.true;
+    expect(bookingNavigation.props('currentStageIndex')).to.eq(0);
+    expect(bookingNavigation.props('maxAllowedStageIndex')).to.eq(0);
+    expect(bookingNavigation.props('production')).to.eq(production);
+    expect(bookingNavigation.props('booking')).to.eq(
+      bookingComponent.vm.booking
+    );
+  });
```
Comment (streth11) [inline tests/unit/views/booking/Book.spec.js:83]: is it woth testing when the sidebar navigation dissapears in mobile?

File: src/fakeApi/index.js [hunk 2]
```diff
@@ -109,19 +254,26 @@ export function makeServer({ environment = 'development' } = {}) {
...
-      server.create('ProductionNode', {
-        name: 'A Default Production',
-      });
+      // /**
+      //  * Fake Performance 4 - A complete random production called A Default Production
+      //  */
+      // server.create('ProductionNode', {
+      //   name: 'A Default Production',
+      // });
+
```
Comment (streth11) [inline src/fakeApi/index.js:271]: why commented?

Comment (streth11) [inline src/fakeApi/index.js:271]: why commented

File: tests/unit/views/booking/Book.spec.js (hunk 1 above)
Comment (atoff) [reply tests/unit/views/booking/Book.spec.js:83]: This is more of a visual thing rather than controlled by code, probably best to leave to E2E testing

**Review by streth11 — APPROVED** [review]
now we are cooking with gas

**Pull request merged**